File: leetcode/001-Two-Sum/py3/TwoSum_001.py
#@author: cchen
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer[]}
    def twoSum(self, nums, target):
        dict = {}
        for i in range(len(nums)):
            if target - nums[i] in dict:
                return [dict[target - nums[i]] + 1, i + 1]
            dict[nums[i]] = i

File: leetcode/001-Two-Sum/js/TwoSum_001.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    var num, i = 0, map = {};
    for (num of nums) {
        if (target - num in map) {
            return [map[target - num] + 1, i + 1];
        }
        map[num] = i;
        i++;
    }
};
